### DISCUSSION TITLE
What is new in FIDO2 (WebAuthn, Passkeys) support between Keycloak 23 and 21?

### DISCUSSION THREAD

**Discussion by christian-2** [keycloak/keycloak#26541] [Q&A] [answered]
Last year I had built a built a [demo](https://github.com/christian-2/oidc-passkey) that employs WebAuthn for SSO with Keycloak 21.1.2. Now the [release notes](https://www.keycloak.org/docs/latest/release_notes/index.html#keycloak-23-0-0) for Keycloak 23.0.0 mention *Passkeys support* as a new feature; Passkeys is a superset of WebAuthn within the umbrella of FIDO2 (broadly speaking).

Hence I am wondering: what is the delta in FIDO2 (WebAuthn, Passkeys, etc.) support between Keycloak 23 and 21? Can somebody please comment and/or is there specific documentation available? I'm aware of the chapter [*Passkeys*](https://www.keycloak.org/docs/latest/server_admin/index.html#passkeys_server_administration_guide) in the [*Server Administration*](https://www.keycloak.org/docs/latest/server_admin/index.html) guide, but at this point I am specifically interested in what is new in relation.

**Comment by thomasdarimont** [comment]
Passkeys based authentication is known to work for a while already, however the Keycloak team never advertised much on it.
For Keycloak 23 the Keycloak maintainers did some testing to verify the current support for passkeys works as intended.

AFAIR passkeys support for conditional UI is currently still WIP of support, see: 
PR: https://github.com/keycloak/keycloak/pull/24305
Discussion: https://github.com/keycloak/keycloak/discussions/15860

**Reply by christian-2** [reply]
@thomasdarimont thanks, it's clear now.

BTW, in relation to that FIDO2 (WebAuthn, Passkeys) demo I had also made an enhancement proposal (informally in discussion #22705) regarding Keycloak as identity provider for VMware. If that is of interest, I could create an issue and contribute a PR.

**Comment by tnorimat** [comment, marked as answer]
@christian-2 Hello, there is [the epic issue](https://github.com/keycloak/keycloak/issues/23656) about passkeys support by Keycloak.
One of the remaining 2 issues is [Passkeys: Supporting WebAuthn Conditional UI](https://github.com/keycloak/keycloak/issues/24264). It is in progress.
The other of the remaning 2 issuess is [passkeys: automated testing](https://github.com/keycloak/keycloak/issues/23659). IMO, there is no passkey specific test because passskey authentication is the same as webauthn authentication technically.
